### PR TITLE
GH527: Added search on enter.

### PR DESCRIPTION
**Pull request by DamianMaslanka5**
Related to #527 

Added checkbox in settings to disable/enable search with delay, by default searching with delay is disabled.

settings

Added search button that is active even if search with delay is enabled.

button

### CODE REVIEW

File: Source/ChocolateyGui/Views/RemoteSourceView.xaml [hunk 1]
```diff
@@ -220,7 +220,11 @@
                            Target="{Binding ElementName=SearchTextBox}" Content="{x:Static lang:Resources.RemoteSourceView_SearchBoxText}"/>
                     <TextBox x:Name="SearchTextBox" Width="200" FontSize="14"
                              VerticalAlignment="Center"
-                             Text="{Binding SearchQuery, UpdateSourceTrigger=PropertyChanged}" />
+                             Text="{Binding SearchQuery, UpdateSourceTrigger=PropertyChanged}" />
+                    <Button Command="{commands:DataContextCommandAdapter LoadPackages, CanLoadRemotePackages}"
+                            Content="Search"
+                            IsDefault="True"
+                            Style="{DynamicResource IconFlatButtonStyle}" />
```
Comment (punker76) [inline Source/ChocolateyGui/Views/RemoteSourceView.xaml]: I would suggest to use the MahApps search TextBox style.

```xaml
<TextBox x:Name="SearchTextBox" Width="200" FontSize="14"
            VerticalAlignment="Center"
            mah:TextBoxHelper.ButtonCommand="{commands:DataContextCommandAdapter LoadPackages, CanLoadRemotePackages}"
            Text="{Binding SearchQuery, UpdateSourceTrigger=PropertyChanged}"
            Style="{DynamicResource SearchMetroTextBox}">
    <TextBox.InputBindings>
        <KeyBinding Command="{commands:DataContextCommandAdapter LoadPackages, CanLoadRemotePackages}" Key="Return" />
    </TextBox.InputBindings>
</TextBox>
```
with
```xaml
xmlns:mah="http://metro.mahapps.com/winfx/xaml/controls"
```

2018-02-05_21h07_02

**Comment by gep13** [comment]
@DamianMaslanka5 The build failure is nothing for you to worry about, that is something that we will need to fix.

**Comment by gep13** [comment]
@punker76 @RichiCoder1 are you both happy with this modification?

If so, it would be good to get this merged in, ready for the next release.

File: Source/ChocolateyGui/ViewModels/RemoteSourceViewModel.cs [hunk 2]
```diff
@@ -330,5 +274,87 @@ protected override void OnInitialize()
...
+
+        protected override void OnInitialize()
+        {
+            try
+            {
+                Observable.FromEventPattern<EventArgs>(_configService, "SettingsChanged")
+                    .ObserveOnDispatcher()
+                    .Subscribe(eventPattern =>
+                    {
+                        var appConfig = (AppConfiguration)eventPattern.Sender;
```
Comment (RichiCoder1) [inline Source/ChocolateyGui/ViewModels/RemoteSourceViewModel.cs]: Smart! I like

**Comment by gep13** [comment]
@DamianMaslanka5 I have fixed the issue that was present in the build, and I have rebased your PR on the latest copy of the develop branch.  I will wait and check that the build is working as expected, and if it does, I will merge this PR.

**Comment by gep13** [comment]
Ah, that's better, a green build 😄

**Comment by gep13** [comment]
@DamianMaslanka5 thank you again for your help in getting this done!

**Pull request merged**